Userland/Services/SpiceAgent/Message.cpp

```diff
@@ -111,4 +111,34 @@ ErrorOr<String> ClipboardRequestMessage::debug_description()
     return builder.to_string();
 }
 
+ErrorOr<ClipboardMessage> ClipboardMessage::read_from_stream(AK::Stream& stream)
+{
+    auto value = TRY(stream.read_value<u32>());
+    if (value >= to_underlying(ClipboardDataType::__End)) {
+        return Error::from_string_literal("Unsupported clipboard type");
+    }
+
+    auto type = static_cast<ClipboardDataType>(value);
+    auto contents = TRY(stream.read_until_eof());
```

Comment:
I might be missing something, but don't we have to length-limit this?

Reply:
No, since the data should fill up the rest of the buffer. Remember, the `AK::Stream` we pass in to this is only as long as the chunk header of the message says it should be :^)

Reply:
Oh, I see, I missed the copying to the intermediary buffer. That's fine then.